pkg/volume/util.go

```diff
@@ -142,8 +142,9 @@ func (c *realRecyclerClient) WatchPod(name, namespace string, stopChannel chan s
 	cache.NewReflector(podLW, &api.Pod{}, queue, 1*time.Minute).RunUntil(stopChannel)
 
 	return func() *api.Pod {
-		obj := queue.Pop()
-		return obj.(*api.Pod)
+		var pod *api.Pod
+		queue.Pop(func(obj interface{}) error { pod = obj.(*api.Pod); return nil })
```

Comment:
We can't really stop returning the object, because controller framework is using to requeue this object conditionally:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/framework/controller.go#L130

We may consider adding another parameter "enqueueOnError bool" to the signature.